enhancements/microshift/etcd-supportability.md

```diff
@@ -0,0 +1,249 @@
+  to communicate with the database in case something goes wrong.
+
+### Goals
+
+* Describe the relationship between MicroShift and etcd and how it
+  differs from OpenShift.
+* Outline the support procedures for the user stories listed above.
+
+### Non-Goals
+
+* This document will not contain complete support procedures. More
+  complete documentation will be written separately.
+* MicroShift does not own the workloads running on it. Backing up and
+  restoring the state of applications running on MicroShift is out of
+  scope.
+
+## Proposal
+
+### Design Considerations
+
+MicroShift is designed to provide a unified user experience. Many
+components, including the kubernetes API server, kubelet, and
+controllers, are compiled into a single binary. Users interact with
+MicroShift, rather than the individual components.
+
+MicroShift is not a “full stack” solution. It does not manage the OS
+on which it runs, and is meant to integrate with, rather than subsume,
+tools and processes for managing the host. “MicroShift is an
+application running on RHEL.”
+
+MicroShift is meant to run on edge devices, where manual operations
+are not always possible. It should therefore be as resilient as
+possible and take automated action when possible.
+
+There is a practical benefit of designing the workflows, processes,
+and tools for MicroShift so that an administrator’s experience with
+OCP can translate to experience with MicroShift. While we do not want
+MicroShift to diverge just for the sake of being different, there are
+already enough differences in operation due to the implementation
+details of MicroShift and the Device Edge product that some divergence
+from OCP may be inevitable.
+
+We do not want to force alignment at the expense of other design
+principles. For example, after recent issues compiling etcd into the
+same binary with the other aspects of MicroShift, it is now being
+delivered as a separate binary. However, to retain the design
+principle that the user should not have to orchestrate multiple
+services to use MicroShift, the MicroShift process manages the etcd
+process for the user. Starting and stopping MicroShift starts and
+stops etcd transparently. This is similar to the experience of using
+k3s or other single-binary distributions of kubernetes, even though
+the details are different.
+
+We need to keep the 4 footprints in mind (RAM, CPU, storage,
+bandwidth). Tools should be small, and included in the OS image along
+with MicroShift rather than delivered via container images at runtime.
+
+We have an aggressive schedule for bringing MicroShift to GA
+status. We therefore want to choose simple solutions, even if that
+means we need to do additional work to improve tools in later
+releases.
+
+### Workflow Description
+
+#### Database backup and restore
+
+For backup and restore we need to consider both the user-driven backup
+and the Greenboot backup scenarios.
```

Comment:
Greenboot automatic restore will need to account for the extra disk space to store previous state, right?

Reply:
It will. We'll cover those details in the greenboot enhancement in #1312